src/js/builtins/JSBufferConstructor.ts

```diff
@@ -1,78 +1,43 @@
 // This is marked as a constructor because Node.js allows `new Buffer.from`,
 // Some legacy dependencies depend on this, see #3638
 $constructor;
-export function from(items) {
-  if ($isUndefinedOrNull(items)) {
-    throw new TypeError(
-      "The first argument must be one of type string, Buffer, ArrayBuffer, Array, or Array-like Object.",
-    );
-  }
+export function from(value, encodingOrOffset, length) {
+  if (typeof value === "string") return new $Buffer(value, encodingOrOffset);
 
-  // TODO: figure out why private symbol not found
-  if (
-    typeof items === "string" ||
-    (typeof items === "object" &&
-      ($isTypedArrayView(items) ||
-        items instanceof ArrayBuffer ||
-        items instanceof SharedArrayBuffer ||
-        items instanceof String))
-  ) {
-    switch ($argumentCount()) {
-      case 1: {
-        return new $Buffer(items);
-      }
-      case 2: {
-        return new $Buffer(items, $argument(1));
-      }
-      default: {
-        return new $Buffer(items, $argument(1), $argument(2));
-      }
+  if (typeof value === "object" && value !== null) {
+    if ($inheritsArrayBuffer(value)) return new $Buffer(value, encodingOrOffset, length);
+    if ($isTypedArrayView(value)) return new $Buffer(value, encodingOrOffset, length);
+
+    const valueOf = value.valueOf && value.valueOf();
+    if (valueOf != null && valueOf !== value && (typeof valueOf === "string" || typeof valueOf === "object")) {
+      return Buffer.from(valueOf, encodingOrOffset, length);
     }
-  }
-  if (typeof items === "object") {
-    const data = items.data;
-    if (items.type === "Buffer" && Array.isArray(data)) {
+
+    if (value.length !== undefined || $inheritsArrayBuffer(value.buffer)) {
```

Comment:
do we need to access .buffer specifically? could we instead check if it's any typed array view and if it isn't, only then check if it inherits arraybuffer? calling .buffer often has a side effect of cloning the contents of the array

Reply:
that happens on line 9

Reply:
i wonder if that check supersedes the need for this if block entirely

Reply:
wait i have a better idea